### PR TITLE
RD-693 Add filters functionality

### PR DESCRIPTION
**Pull request by YoniItzhak**
This PR implements the main logic of filters. Meaning, the conditions added to the query based on the provider filter rules.
I would like to get a review on the way I implemented it, there might be a more efficient way.

### CODE REVIEW

File: rest-service/manager_rest/storage/filters.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        '<key>=[<value1>,<value2>,...], <key>!=<value>, '
+        '<key>!=[<value1>,<value2>,...], <key> is null, '
+        '<key> is not null'.format(labels_filter_value)
+    )
+
+
+def _get_label_value(raw_label_value):
+    # raw_label_value should be of the form [<value1>,<value2>,...]
+    # in order to count as a list
+    match_list = re.match(r'^\[(\S+)\]$', raw_label_value)
```
Comment (YoniItzhak) [inline rest-service/manager_rest/storage/filters.py:72]: This means that this list `[b, c]` is invalid, due to the space.

File: rest-service/manager_rest/storage/filters.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                elif match_not_null:
+                    query = query.filter(
+                        key_exist(labels_model, match_not_null.group(1)))
+                else:
+                    _raise_bad_labels_filter(labels_filter)
+
+            else:
+                _raise_bad_labels_filter(labels_filter)
+
+        except SyntaxError:
```
Comment (tehasdf) [inline rest-service/manager_rest/storage/filters.py:46]: nothing in here raises a SyntaxError i dont think, the only thing that even would, was the ast

Comment (YoniItzhak) [reply rest-service/manager_rest/storage/filters.py]: Correct, this was a leftover.

File: rest-service/manager_rest/storage/filters.py (hunk 1 above)
Comment (geokala) [reply rest-service/manager_rest/storage/filters.py:61]: Shouldn't this be r'^\[([^\]]+)\]$'?

Comment (YoniItzhak) [reply rest-service/manager_rest/storage/filters.py:61]: Right, this makes more sense, changing.

File: rest-service/manager_rest/test/endpoints/test_filters.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+from manager_rest.storage.filters import add_labels_filters_to_query
+from manager_rest.storage.resource_models import Deployment, DeploymentLabel
+
+
+class FiltersFunctionalityTest(base_test.BaseServerTestCase):
+
+    LABELS = [{'a': 'b'}, {'a': 'z'}, {'c': 'd'}]
+    LABELS_2 = [{'a': 'b'}, {'c': 'z'}, {'e': 'f'}]
+
+    def test_filters_functionality(self):
```
Comment (geokala) [inline rest-service/manager_rest/test/endpoints/test_filters.py:12]: Is it worth adding a test to ensure that we complain if passed an invalid label in a filter expression?

Comment (YoniItzhak) [reply rest-service/manager_rest/test/endpoints/test_filters.py:12]: Of course, my bad, adding.

**Review by tehasdf — APPROVED** [review]

**Pull request merged**